source/common/tcp_proxy/upstream.cc

```diff
@@ -222,7 +203,17 @@ bool HttpConnPool::valid() const { return conn_pool_ != nullptr; }
 
 void HttpConnPool::newStream(GenericConnectionPoolCallbacks* callbacks) {
   callbacks_ = callbacks;
-  upstream_ = std::make_unique<HttpUpstream>(upstream_callbacks_, hostname_);
+  switch (type_) {
+  case Http::CodecClient::Type::HTTP1:
+    upstream_ = std::make_unique<Http1Upstream>(upstream_callbacks_, hostname_);
+    break;
+  case Http::CodecClient::Type::HTTP2:
+    upstream_ = std::make_unique<Http2Upstream>(upstream_callbacks_, hostname_);
+    break;
+  case Http::CodecClient::Type::HTTP3:
+  default:
```

Comment:
can we add validation in valid() that type is either HTTP1 or HTTP2?
Then here we can just say
if (http1) {
} else {
}
and not have uncovered crashy code.

Reply:
ping?